### ISSUE TITLE
Simplify controller functions

### ISSUE THREAD

**Issue by jpahm** [UTDNebula/nebula-api#204]
Currently the controller functions are written in an over-complicated format, like so:
```go
func CourseAll() gin.HandlerFunc {
	return func(c *gin.Context) {
		// Controller code here...
	}
}
```
This is over-complicated and cluttered, since the above code could be easily rewritten like so:
```go
func CourseAll(c *gin.Context) {
	// Controller code here...
}
```

We should seek to simplify all of the controller code like shown -- code using the controllers will need to be adjusted accordingly.

**Comment by mikehquan19** [comment]
Hi! I'm new to Go and Mongo. Can I get assigned to this task ?

**Comment by jpahm** [comment]
> Hi! I'm new to Go and Mongo. Can I get assigned to this task ?

Yes you can, it's all yours!

**Comment by jpahm** [comment]
Closed via #219
